fate/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -60,14 +55,39 @@
   private final Lock cacheReadLock = cacheLock.readLock();
 
   private final HashMap<String,byte[]> cache;
-  private final HashMap<String,Stat> statCache;
+  private final HashMap<String,ZcStat> statCache;
   private final HashMap<String,List<String>> childrenCache;
 
   private final ZooReader zReader;
 
+  public static class ZcStat {
+    private long emphemeralOwner;
+
+    public ZcStat() {
+
+    }
+
+    private ZcStat(Stat stat) {
+      this.emphemeralOwner = stat.getEphemeralOwner();
+    }
+
+    public long getEphemeralOwner() {
+      return emphemeralOwner;
+    }
+
+    private void set(ZcStat cachedStat) {
+      this.emphemeralOwner = cachedStat.emphemeralOwner;
+    }
+
+    @VisibleForTesting
```

Comment:
That's great that this is visible only for testing, but I don't understand what the relationship is between "ephemheralOwner" and "session". Neither the method nor the test which calls it explains this.

Reply:
oh, alot of code uses the ephemheralOwner as a session id.  However here it should not be called session.